src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -199,10 +199,7 @@ internal void AddGenerators(ImmutableArray<ISourceGenerator>.Builder builder, st
         private static AnalyzerLoadFailureEventArgs CreateAnalyzerFailedArgs(Exception e, string? typeName = null)
         {
             // unwrap:
-            if (e is { InnerException: FileLoadException inner })
-            {
-                e = inner;
-            }
+            e = (e as TargetInvocationException) ?? e;
```

Comment:
Does this line do anything?

Reply:
I don't think it does. I'd like to remove it separately. Feel free to open a PR if you have a minute.

Reply:
https://github.com/dotnet/roslyn/pull/62508